src/bundle/Controller/UniversalDiscoveryController.php

```diff
@@ -226,11 +232,15 @@ private function getParentLocationPath(Location $location): array
         }, $parentPath);
     }
 
-    private function getBreadcrumbLocations(Location $location): array
-    {
-        $filter = $location->id === 1
-            ? new Query\Criterion\ParentLocationId($location->path)
-            : new Query\Criterion\LocationId($this->getParentLocationPath($location));
+    private function getBreadcrumbLocations(
+        int $locationId,
+        int $rootLocationId = self::ROOT_LOCATION_ID
+    ): array {
+        $filter = $locationId === $rootLocationId
+            ? new Query\Criterion\ParentLocationId($rootLocationId)
+            : new Query\Criterion\LocationId($this->getParentLocationPath(
+                $this->locationService->loadLocation($locationId)
+            ));
```

Comment:
IMHO you could introduce QueryType here.

Reply:
This won't work. QueryTypes are meant for Content not for Location search. Here the only optimization would be creating a new class extending `Query\Criterion\LocationId` to hide work done in `$this->getParentLocationPath()` but it isn't going to be that reusable,